internal/resources/iampolicy/scope.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package iampolicy
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	clustermodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/cluster"
+	clustergroupmodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/clustergroup"
+	namespacemodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/namespace"
+	organizationmodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/organization"
+	workspacemodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/workspace"
+)
+
+type scopeType int
+
+const (
+	unknown scopeType = iota
+	organization
+	clusterGroup
+	cluster
+	workspace
+	namespace
+)
+
+var (
+	stringToScope = map[string]scopeType{
+		orgKey:          organization,
+		clusterGroupKey: clusterGroup,
+		clusterKey:      cluster,
+		workspaceKey:    workspace,
+		namespaceKey:    namespace,
+	}
+)
+
+func (s scopeType) String() string {
+	return [...]string{
+		"unknown",
```

Comment:
when is the scope `unknown`?

Reply:
When user gives any other input apart from the accepted scopes.

Reply:
That is not acceptable. The allowed blocks should be defined in the schema, and the validation function should handle this. Having a scope type `unknown` is not intuitive.